server/vulnerabilities/msrc/parser.go

```diff
@@ -0,0 +1,153 @@
+package msrc
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	msrc_parsed "github.com/fleetdm/fleet/v4/server/vulnerabilities/msrc/parsed"
+	msrc_xml "github.com/fleetdm/fleet/v4/server/vulnerabilities/msrc/xml"
+)
+
+func parseFeed(feedFilePath string) (map[string]*msrc_parsed.SecurityBulletin, error) {
+	r, err := os.Open(feedFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("msrc parser: %w", err)
+	}
+	defer r.Close()
+
+	feedResultXML, err := parseXML(r)
+	if err != nil {
+		return nil, fmt.Errorf("msrc parser: %w", err)
+	}
+
+	bulletins, err := mapToSecurityBulletins(feedResultXML)
+	if err != nil {
+		return nil, fmt.Errorf("msrc parser: %w", err)
+	}
+
+	return bulletins, nil
+}
+
+func mapToSecurityBulletins(rXML *msrc_xml.FeedResult) (map[string]*msrc_parsed.SecurityBulletin, error) {
+	// We will have one bulletin for each product.
+	bulletins := make(map[string]*msrc_parsed.SecurityBulletin)
+	pIDToPName := make(map[string]string)
+
+	for pID, p := range rXML.WinProducts {
+		name := msrc_parsed.NewFullProductName(p.FullName).Name()
```

Comment:
I commented above about using `parsed.NewProduct` or similar, this line is what triggered that thought.